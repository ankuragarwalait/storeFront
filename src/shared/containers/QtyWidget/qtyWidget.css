.counter-field, .counter-control {
    display: inline-block;
    vertical-align: middle;
}

.counter-field {
    width: .5rem;
    padding: 1.25rem; 
    margin-right: .25rem;
}

.counter-control button {
    background-color: grey;
    padding: .3rem .6rem;
    display: block;
    border: none;
    margin-bottom: .25rem;
}

.counter-control .btn-dec {
    padding: .4rem .65rem;
    margin-bottom: 0;
}