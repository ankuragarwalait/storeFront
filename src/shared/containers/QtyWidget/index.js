import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './qtyWidget.css';

export default class QtyWidget extends Component {
    static propTypes = {
        shouldAutoUpdate: PropTypes.bool.isRequired,
        updateQuantity: PropTypes.func,
        prodQty: PropTypes.number,
        qtyRef: PropTypes.shape({ current: PropTypes.instanceOf(Element) })
    }
    
    constructor(props) {
        super(props);
        this.state = {
            count : this.props.prodQty || 1
        }
        this.increment = this.increment.bind(this);
        this.decrement = this.decrement.bind(this);
    }

    increment() {
        this.setState({
            count: this.state.count + 1
        })
    }

    decrement() {
        this.setState({
            count: (this.state.count - 1) || 1
        })
    }

    componentDidUpdate() {
        if(this.props.shouldAutoUpdate) {
            this.props.updateQuantity(this.state.count, this.props.prodId);
        }
    }

    render() {
        return (
            <div>
                <input className="counter-field" value={this.state.count} readOnly ref={this.props.qtyRef}/>
                <div className="counter-control">
                    <button onClick={this.increment}>+</button>
                    <button className="btn-dec" onClick={this.decrement}>-</button>
                </div>
            </div>
        )
    }
}