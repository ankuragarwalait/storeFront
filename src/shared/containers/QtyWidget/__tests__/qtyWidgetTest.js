import renderer from 'react-test-renderer';
import React from 'react';
import QtyWidget from '..';

describe('test qty widget snapshots', () => {
    it('qty widget render snapshot', ()=> {
        const props = {
            prodId: 'mashiko-yaki-indigo-small-plate',
            qtyRef: React.createRef(),
            shouldAutoUpdate: true,
            updateQuantity: jest.fn()
        }
        const tree = renderer.create(<QtyWidget {...props}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});

describe('test qty widget event handlers', ()=> {
    it('test increment and auto update quantity', ()=> {
        const props = {
            prodQty: 3,
            prodId: 'mashiko-yaki-indigo-small-plate',
            qtyRef: React.createRef(),
            shouldAutoUpdate: true,
            updateQuantity: jest.fn()
        }

        const tree = renderer.create(<QtyWidget {...props}/>);

        tree.root.findAllByType('button')[0].props.onClick();
        expect(tree.root.findByType('input').props.value).toBe(4);
        expect(tree.root.props.updateQuantity).toHaveBeenCalled();
    });

    it('test decrement without calling auto update', ()=> {
        const props = {
            prodQty: 1,
            prodId: 'mashiko-yaki-indigo-small-plate',
            qtyRef: React.createRef(),
            shouldAutoUpdate: false,
            updateQuantity: jest.fn()
        }

        const tree = renderer.create(<QtyWidget {...props}/>);

        tree.root.findAllByType('button')[1].props.onClick();
        expect(tree.root.findByType('input').props.value).toBe(1);
        expect(tree.root.props.updateQuantity).not.toHaveBeenCalled();
    });

});