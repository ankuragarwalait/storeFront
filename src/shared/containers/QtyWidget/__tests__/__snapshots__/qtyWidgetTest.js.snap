// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test qty widget snapshots qty widget render snapshot 1`] = `
<div>
  <input
    className="counter-field"
    readOnly={true}
    value={1}
  />
  <div
    className="counter-control"
  >
    <button
      onClick={[Function]}
    >
      +
    </button>
    <button
      className="btn-dec"
      onClick={[Function]}
    >
      -
    </button>
  </div>
</div>
`;
